### PR TITLE
Support etcd

### PR DESCRIPTION
**Pull request by SunSince90**
This pull requests bring support for the popular distributed key-value store *etcd* to the cnwan-reader.

To use etcd as a service registry, the command `etcd watch etcd [flags]` must be used and, as always, the cnwan-reader will react to all events happening on the cluster.

Please take a look at #18 to track the progress of this pr.

### CODE REVIEW

**Comment by SunSince90** [comment]
Note: the tests just fail because op#25 is not merged, yet

File: pkg/cmd/watch/etcd/fake_kv.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
+// Copyright © 2020 Cisco
```
Comment (ljakab) [inline pkg/cmd/watch/etcd/fake_kv.go:1]: s/2020/2021/

File: pkg/cmd/watch/etcd/watcher_test.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
+// Copyright © 2020 Cisco
```
Comment (ljakab) [inline pkg/cmd/watch/etcd/watcher_test.go:1]: s/2020/2021/

File: pkg/cmd/watch/etcd/fake_queue.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
+// Copyright © 2020 Cisco
```
Comment (ljakab) [inline pkg/cmd/watch/etcd/fake_queue.go:1]: s/2020/2021/

File: docs/docker_usage.md [hunk 4]
```diff
@@ -123,3 +123,23 @@ docker run \
...
 cnwan/cnwan-reader poll cloudmap --conf ./options/conf.yaml
 ```
+
+### With etcd
+
+In the following example, the CN-WAN Reader watches changes in etcd with the following requirements:
+
+* The *allowed* services have at least the `cnwan.io/traffic-profile` key in their metadata.
+* Username is `admin` and password `s5B7&$n_12C`.
+* There is only on etcd node, on address `10.11.12.13` on default port (`2379`).
```
Comment (ljakab) [inline docs/docker_usage.md:133]: s/on/one/

File: docs/usage.md [hunk 5]
```diff
@@ -190,3 +201,23 @@ or just:
...
 cnwan-reader --conf /path/to/configuration/file.yaml
 ```
+
+### With etcd
+
+In the following example, the CN-WAN Reader watches changes in etcd with the following requirements:
+
+* The *allowed* services have at least the `cnwan.io/traffic-profile` key in their metadata.
+* Username is `admin` and password `s5B7&$n_12C`.
+* There is only on etcd node, on address `10.11.12.13` on default port (`2379`).
```
Comment (ljakab) [inline docs/usage.md:211]: s/on/one/

**Review by ljakab — CHANGES_REQUESTED** [review]
Elis, this looks very good, just a few small typo corrections/copyright year changes needed

File: go.mod [hunk 6]
```diff
@@ -8,8 +8,17 @@ require (
...
+	google.golang.org/genproto v0.0.0-20201210142538-e3217bee35cc
+	google.golang.org/grpc v1.34.0 // indirect
 	gopkg.in/yaml.v2 v2.2.8
 )
+
+replace (
+	github.com/coreos/etcd => github.com/coreos/etcd v3.3.25+incompatible
+	go.etcd.io/bbolt => go.etcd.io/bbolt v1.3.5
+	go.etcd.io/etcd => go.etcd.io/etcd v0.5.0-alpha.5.0.20200910180754-dd1b699fc489 // ae9734ed278b is the SHA for git tag v3.4.13
+	google.golang.org/grpc => google.golang.org/grpc v1.27.1
```
[the remark this answers is not in the document]
Comment (SunSince90) [reply go.mod:26]: Unfortunately, as long as etcd team does not provide a good/stable module support, this will stay like this

**Pull request merged**